include/dlaf/sender/transform_mpi.h

```diff
@@ -41,42 +42,20 @@ void consumeCommunicatorWrapper(T&) {}
 /// callable. The wrapper then waits for the request to complete with
 /// yield_while.
 ///
-/// This could in theory be a lambda inside transformMPI.  However, clang at
-/// least until version 12 fails with an internal compiler error with a trailing
-/// decltype for SFINAE. GCC has no problems with a lambda.
```

Comment:
Did you test that this comment no longer applies?

Reply:
I'm still wondering about the removal of the comment.

Reply:
I played around, but would prefer to leave this until a later PR

Reply:
That's ok, but then why is the comment removed?